### DISCUSSION TITLE
Focus not working on error (useFieldArray + yup + FormProvider + trigger)

### DISCUSSION THREAD

**Discussion by andyg-12** [react-hook-form/react-hook-form#6453] [Q&A] [answered]
Focus is working for most of the cases but it doesn't work when I use **useFieldArray**. (v7)

CSB: https://codesandbox.io/s/react-hook-form-focus-forked-mhn1n?file=/src/SectionArray.js

Steps to reproduce:

1. Type 'AAA' for MUI Textfield
2. Select any option from the dropdown
3. Click Submit or Trigger (I added both in case they're different)

I've inspected a similar working example (v6): https://github.com/react-hook-form/react-hook-form/discussions/5891#discussioncomment-991807

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/issues/6455
looking into it.

**Comment by bluebill1049** [comment, marked as answer]
issue is due to ref assignment: https://github.com/react-hook-form/react-hook-form/issues/6455
